### DISCUSSION TITLE
Adding OOP Background Task  to WinUI3 / WindowsAppSDK App

### DISCUSSION THREAD

**Discussion by myokeeh** [microsoft/CsWinRT#1243] [Q&A]
I'm converting a UWP app to WinUI3/WindowsAppSDK and converting the OOP background task with the help of this sample: https://github.com/microsoft/CsWinRT/tree/master/src/Samples/BgTaskComponent

I have several errors similar to the one below. How do I fix this?

````
Severity	Code	Description	Project	File	Line	Suppression State
Error	APPX1101	Payload contains two or more files with the same destination path 'WinRT.Host.runtimeconfig.json'. Source files: 
C:\AppBackgroundTasks\WinRT.Host.runtimeconfig.json
C:\AppBackgroundTasks\bin\x64\Debug\net6.0-windows10.0.22621.0\WinRT.Host.runtimeconfig.json	AppWTS	C:\Users\hhhhh\.nuget\packages\microsoft.windowsappsdk\1.1.4\buildTransitive\Microsoft.Build.Msix.Packaging.targets	1479	

````

**Comment by j0shuams** [comment]
Hi @myokeeh, that sample from the C#/WinRT repo is outdated. Primarily, we have since added automation of the `WinRT.Host.runtimeconfig.json`. My best guess from the error message is you have written one yourself (as the Sample suggests), and the automated logic is also making one. If you did write one, would you try deleting it (and the [logic to place in the output](https://github.com/microsoft/CsWinRT/blob/fa7f5565cb53353dc15c28a09ebd500577dc0b72/src/Samples/BgTaskComponent/BgTaskComponent/BgTaskComponent.csproj#L19-L23)) and rebuilding?

**Reply by myokeeh** [reply]
I actually just copied the sample project and changed the .cs file. Do you mean delete the `WinRT.Host.runtimeconfig.json` file? I just tried that and getting similar APPX1101 errors.

csproj:
```
<Project Sdk="Microsoft.NET.Sdk">
  
  <PropertyGroup>
    <TargetFramework>net6.0-windows10.0.22621.0</TargetFramework>
    <Platforms>x64;x86</Platforms>
    <RuntimeIdentifiers>win10-x86;win10-x64</RuntimeIdentifiers>
  </PropertyGroup>

  <!-- CsWinRT properties -->
  <PropertyGroup>
    <CsWinRTComponent>true</CsWinRTComponent>
    <CsWinRTWindowsMetadata>10.0.22621.0</CsWinRTWindowsMetadata>
    <SupportedOSPlatformVersion>10.0.22621.0</SupportedOSPlatformVersion>
  </PropertyGroup>
  <ItemGroup>
    <None Remove="WinRT.Host.runtimeconfig.json" />
  </ItemGroup>

  <ItemGroup>
    <PackageReference Include="CommunityToolkit.WinUI.Connectivity" Version="7.1.2" />
    <PackageReference Include="CommunityToolkit.WinUI.Notifications" Version="7.1.2" />
    <PackageReference Include="Microsoft.AppCenter.Analytics" Version="4.5.3" />
    <PackageReference Include="Microsoft.AppCenter.Crashes" Version="4.5.3" />
    <PackageReference Include="Microsoft.Azure.Mobile.Client" Version="4.2.0" />
    <PackageReference Include="Microsoft.Graph" Version="4.36.0" />
    <PackageReference Include="Microsoft.Windows.CsWinRT" Version="2.0.0" />
  </ItemGroup>

  <!--ItemGroup>
    <None Update="C:\Users\Carlo\.nuget\packages\microsoft.windows.cswinrt\1.6.5\hosting\x64\native\WinRT.Host.dll">
      <CopyToOutputDirectory>Never</CopyToOutputDirectory>
    </None>
  </ItemGroup-->

</Project>
```

WinRT.Host.runtimeconfig.json:
```
{
    "runtimeOptions": {
        "tfm": "net6.0",
        "rollForward": "LatestMinor",
        "framework": {
            "name": "Microsoft.NETCore.App",
            "version": "6.0.0"
        }
    }
}
```

This is what's on the csproj:
![image](https://user-images.githubusercontent.com/3663297/185496417-0f619ff1-0e09-4342-92fb-8a664bfdaf55.png)

![image](https://user-images.githubusercontent.com/3663297/185497122-4341406f-ef6f-423c-b231-64bc750142a1.png)

**Reply by j0shuams** [reply]
Hi I suggest relying on the built-in logic for WinRT.Host.runtimeconfig.json, and by that I mean remove the `None` line you have for it.

**Reply by myokeeh** [reply]
I think that would put it back to how the sample code is setup, but I just made that change. I'm still getting the same errors. FYI, the background task is a separate csproj from the app.

**Comment by nickrandolph** [comment]
Link to WinAppSdk repo - https://github.com/microsoft/WindowsAppSDK/discussions/2859

**Comment by GreenShadeZhang** [comment]
At present, I also report some errors when I use background tasks to package msix in wasdk, and I recommend that the official give a sample demo to ensure that the service can be used normally to ensure that it is not a WASDK bug.

**Comment by Xyncgas** [comment]
Can you please let me know if background task cam be used to detect when application is now in background instead of focused / running in foreground, since you are familiar with the subject, I am really interested to understand.

Is background task a cross platform api?

**Comment by myokeeh** [comment]
@Scottj1s, any ideas how I can get this background task to work?

**Reply by myokeeh** [reply]
I have some success here https://github.com/microsoft/CsWinRT/issues/1327, but still have APPX1101 errors.
